resources/apigateway/gateway.yaml

```diff
@@ -0,0 +1,43 @@
+#
+# Add gateway API resources to enable HTTPS traffic into the cluster
+#
+apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  name: kong
+  annotations:
+    konghq.com/gatewayclass-unmanaged: 'true'
+spec:
+  controllerName: konghq.com/kic-gateway-controller
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: kong-gateway
+spec:
+  gatewayClassName: kong
+  listeners:
+  - name: https-democluster
+    port: 443
+    protocol: HTTPS
+    hostname: "*.democluster.example"
+    allowedRoutes:
+      namespaces:
+        from: 'All'
+    tls:
+      mode: Terminate
+      certificateRefs:
+        - kind: Secret
+          name: external-tls
```

Comment:
All of the certificate logic is managed in one place.

One thing I learned recently:
- Renewal of the default SSL certificate is not detected automatically by Kong
- But renewal of Kubernetes secrets updates other YAML resources on the fly

So if the cert was issued by a provider like Lets Encrypt, the gateway cert renewal would be automatic. So good to fit in with that I think.